controller.go

```diff
@@ -198,7 +199,7 @@ func (c *controller) RegisterDriver(networkType string, driver driverapi.Driver,
 // NewNetwork creates a new network of the specified network type. The options
 // are network specific and modeled in a generic way.
 func (c *controller) NewNetwork(networkType, name string, options ...NetworkOption) (Network, error) {
-	if name == "" {
+	if name == "" || strings.Contains(name, ".") {
```

Comment:
My point was that this seemed a validation strictly driven by the need of the cli. 
While the backend should enforce a name policy for all, not only network, resources it owns on its own.
The change of having a generic function to do so is much better. We can later put a proper regex inside there or load it from config file. And use that function to validate other resources (like endpoint) later.
Also I am thinking the way libnetwork is being designed as being an indipendent component which can potentially be used by other container based projects, a hard-coded resource name restriction driven by the docker ui behavior might not be the best choice.

But your new change makes it at least easier to cleanly change this later on.

So LGTM.